### ISSUE TITLE
[Bug] `vue/no-export-in-script-setup` not working for TypeScript type exporting

### ISSUE THREAD

**Issue by wenfangdu** [vuejs/eslint-plugin-vue#2108]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.34.0
- **eslint-plugin-vue version:** 9.9.0
- **Node version:** 16.19.1
- **Operating System:** macOS 13

**What did you do?**

```vue
<script setup lang="ts">
  import { RouterLink, RouterView } from 'vue-router'
  import HelloWorld from './components/HelloWorld.vue'

  export type A = string
  export const a = 1
</script>
```

**What did you expect to happen?**

`export type A = string` throws.

**What actually happened?**

`export type A = string` doesn't throw.

**Repository to reproduce this issue**

https://github.com/wenfangdu/vite-vue-3/tree/vue/no-export-in-script-setup

**Comment by ota-meshi** [comment]
I think `export type` is available in SFC. So I think the rule is working correctly.

https://sfc.vuejs.org/#eNotjksOgzAMRK9ieV2SPQIk7uENRSkNahIrNv0IcfcmLTt77HkzO47M5rk5bLGTOXtWEKcbw2OKS0+oQjhQBHBvTllBP+xghB5Es48Lxc7+bQNe0If604SJzSopFuZerXQeCqqFn1K1Elp3wrsqS2ut3ObaZBWT8mLLZPIW1QdnnITmmtNLXC5gwoo4KB54fAGhMEHZ
